src/map_extras.cpp

```diff
@@ -1399,17 +1453,130 @@ static void mx_fumarole( map &m, const tripoint &abs_sub )
 
 static void mx_portal_in( map &m, const tripoint &abs_sub )
 {
-    int x = rng( 5, SEEX * 2 - 6 ), y = rng( 5, SEEY * 2 - 6 );
-    m.add_field( {x, y, abs_sub.z}, fd_fatigue, 3, 0_turns );
-    fungal_effects fe( *g, m );
-    for( int i = x - 5; i <= x + 5; i++ ) {
-        for( int j = y - 5; j <= y + 5; j++ ) {
-            if( rng( 1, 9 ) >= trig_dist( x, y, i, j ) ) {
-                fe.marlossify( tripoint( i, j, abs_sub.z ) );
-                if( one_in( 15 ) ) {
-                    m.place_spawns( GROUP_NETHER_PORTAL, 1, i, j, i, j, 1, true );
+    const tripoint portal_location = { rng( 5, SEEX * 2 - 6 ), rng( 5, SEEX * 2 - 6 ), abs_sub.z };
+    const int x = portal_location.x;
+    const int y = portal_location.y;
+
+    switch( rng( 5, 5 ) ) {
+        //Mycus spreading through the portal
+        case 1: {
+            m.add_field( portal_location, fd_fatigue, 3 );
+            fungal_effects fe( *g, m );
+            for( const auto &loc : g->m.points_in_radius( portal_location, 5 ) ) {
+                if( one_in( 3 ) ) {
+                    fe.marlossify( loc );
                 }
             }
+            //50% chance to spawn pouf-maker
+            m.place_spawns( GROUP_FUNGI_FUNGALOID, 2, x - 1, y - 1, x + 1, y + 1, 1, true );
+            break;
+        }
+        //Netherworld monsters spawning around the portal
+        case 2: {
+            m.add_field( portal_location, fd_fatigue, 3 );
+            for( int i = x - 5; i <= x + 5; i++ ) {
```

Comment:
Why are you using an explicit iteration over x,y here instead of the `points_in_radius` like above?

Reply:
It is a left-over from a previous implementation, I simply didn't change it.

Reply:
Fixed.